src/viewport.js

```diff
@@ -2,13 +2,67 @@
 * Check whether an element is in the viewport by
 * more than offset px.
 */
-export function inViewport (element, offset = 0) {
+export function inViewport (el, offset = 0) {
+    var { top, right, bottom, left } = getVisiblePixels(el, offset);
 
-    let { top, right, bottom, left } = element.getBoundingClientRect();
+    return top     - offset > 0
+        && bottom  - offset > 0
+        && left    - offset > 0
+        && right   - offset > 0
+}
+
+export function getVisiblePixels(el) {
```

Comment:
`getVisiblePixels` measures the amount of visible px on all sides + it also measures visible pixels on each `x` & `y` axis.

This way I can **test** separately which part is cut off. (see updated tests)